### PR TITLE
Fix issue #1829 - better error messages for some wrong uses of direct…

### PR DESCRIPTION
**Pull request by jafingerhut**
…_meter

### CODE REVIEW

**Comment by jafingerhut** [comment]
These are some alternate proposed changes for issue #1829.  I have tested that the error messages seems reasonable when run with `p4c --target bmv2 --arch v1model myprog.p4` for all 4 of the test programs added, but those error messages do not show up in the expected outputs here, I believe because all of the `-stderr` files are stderr outputs from `p4test`.

**Comment by jafingerhut** [comment]
Is there a way to have a -bmv2.p4 source file, so that the BMv2 v1model back end is run, and have an expected stderr output file for them?

**Comment by hesingh** [comment]
If I understand correctly what you want, from `p4c/build/` run this example command:

`bmv2/testdata/p4_16_samples/issue1829-1-bmv2.p4.test`

Then, if any error exists, the error should be emitted to the console and the bmv2 backend is run.

Or try `.\p4c-bm2-ss --std p4-16 ../testdata/p4_16_samples/issue1829-1-bmv2.p4`

A place to add a new negative test for bmv2 is `testdata/p4_16_bmv_errors/`

**Comment by jafingerhut** [comment]
Thanks for the help.  Moving 3 of the tests that should give error messages in the v1model back end code into the p4_16_bmv_errors directory seems to do the trick.

**Comment by mbudiu-vmw** [comment]
If you are happy about the errors produced by this we should merge it.

**Comment by jafingerhut** [comment]
@antoninbas @cc10512 Do one of you want to look over the new error messages introduced by these changes to see if you have any suggested changes?  The patch is short, and the quickest way to see what the new error messages look like.

File: backends/bmv2/common/metermap.cpp [hunk 1]
```diff
@@ -87,7 +93,14 @@ void DirectMeterMap::setDestination(const IR::IDeclaration* meter,
  */
 void DirectMeterMap::setSize(const IR::IDeclaration* meter, unsigned size) {
     auto info = getInfo(meter);
-    CHECK_NULL(info);
+    if (info == nullptr) {
+        /* This case may is reached if a table has a direct_meter
```
Comment (antoninbas) [inline backends/bmv2/common/metermap.cpp]: typo: `may be`

**Comment by jafingerhut** [comment]
Three of the new test programs only produce errors when you run the bmv2 v1model back end, and are in the `testdata/p4_16_bmv_errors` directory.  The p4c test infrastructure expects those to fail, but as far as I can tell does not check the error output against any file contents.

You can check out my branch locally, build p4c, and try running `p4c --target bmv2 --arch v1model prog.p4` on those 3 programs if you want to see the error messages "live", or you can simply read the C++ code changes to see what those error messages look like.

File: backends/bmv2/common/metermap.cpp (hunk 1 above)
Comment (jafingerhut) [reply backends/bmv2/common/metermap.cpp]: fixed

**Comment by jafingerhut** [comment]
@mbudiu-vmw Antonin approved after reviewing error messages, so going ahead and merging this.

**Pull request merged**